diff -rupN flycast2021-lr.orig/core/rend/gles/postprocess.cpp flycast2021-lr/core/rend/gles/postprocess.cpp
--- flycast2021-lr.orig/core/rend/gles/postprocess.cpp	2025-01-16 12:18:25.000000000 -0500
+++ flycast2021-lr/core/rend/gles/postprocess.cpp	2025-05-15 02:39:30.154054635 -0400
@@ -241,7 +241,7 @@ std::array<PostProcessShader, 8> PostPro
 
 void PostProcessor::Init()
 {
-	/*
+	
 	this->width = screen_width;
 	this->height = screen_height;
 
@@ -293,11 +293,11 @@ void PostProcessor::Init()
 		glDisableVertexAttribArray(3);
 	}
 #endif
-	glCheck();*/
+	glCheck();
 }
 
 void PostProcessor::Term()
-{/*
+{
 	glcache.DeleteTextures(1, &texture);
 	texture = 0;
 	glDeleteFramebuffers(1, &framebuffer);
@@ -312,11 +312,11 @@ void PostProcessor::Term()
 #endif
 	vertexArray = 0;
 	PostProcessShader::term();
-	glCheck();*/
+	glCheck();
 }
 
 void PostProcessor::SelectFramebuffer()
-{/*
+{
 	if (framebuffer == 0)
 		Init();
 	glBindFramebuffer(GL_FRAMEBUFFER, framebuffer);
@@ -351,5 +351,5 @@ void PostProcessor::Render(GLuint output
 
    glcache.ClearColor(0.f, 0.f, 0.f, 0.f);
    glClear(GL_COLOR_BUFFER_BIT);
-   glDrawArrays(GL_TRIANGLE_STRIP, 0, 4);*/
+   glDrawArrays(GL_TRIANGLE_STRIP, 0, 4);
 }
 