diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7e135490a..000000000 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -474,6 +474,7 @@
 set_option(SDL_VIVANTE             "Use Vivante EGL video driver" ${UNIX_SYS})
+set_option(SDL_MALI                "Use Mali EGL fbdev video driver" ${UNIX_SYS})

@@ -1501,6 +1502,7 @@
     CheckVivante()
+    CheckMali()


diff --git a/cmake/sdlchecks.cmake b/cmake/sdlchecks.cmake
index a23b2ef07..000000000 100644
--- a/cmake/sdlchecks.cmake
+++ b/cmake/sdlchecks.cmake
@@ -841,6 +841,40 @@ macro(CheckVivante)
 endmacro()

+macro(CheckMali)
+  if(SDL_MALI)
+
+    check_c_source_compiles("
+        #define LINUX
+        #define EGL_API_FB
+        #include <EGL/egl.h>
+        int main(int argc, char** argv) { return 0; }"
+        HAVE_VIDEO_MALI_EGL_FB)
+
+    if(HAVE_VIDEO_MALI_EGL_FB)
+
+      set(SDL_VIDEO_DRIVER_MALI 1)
+      set(HAVE_SDL_VIDEO TRUE)
+
+      file(GLOB MALI_SOURCES
+        ${CMAKE_CURRENT_SOURCE_DIR}/src/video/mali-fbdev/*.c
+      )
+
+      list(APPEND SDL_VIDEO_SOURCES ${MALI_SOURCES})
+
+      add_definitions(-DLINUX -DEGL_API_FB)
+
+      list(APPEND EXTRA_LIBS EGL)
+
+    endif()
+
+  endif()
+endmacro()


diff --git a/include/SDL_config.h.cmake b/include/SDL_config.h.cmake
index ffcafd895..000000000 100644
--- a/include/SDL_config.h.cmake
+++ b/include/SDL_config.h.cmake
@@ -418,6 +418,7 @@
 #cmakedefine SDL_VIDEO_DRIVER_VIVANTE @SDL_VIDEO_DRIVER_VIVANTE@
+#cmakedefine SDL_VIDEO_DRIVER_MALI @SDL_VIDEO_DRIVER_MALI@


diff --git a/src/video/SDL_egl.c b/src/video/SDL_egl.c
index a89c0e812..000000000 100644
--- a/src/video/SDL_egl.c
+++ b/src/video/SDL_egl.c
@@ -66,8 +66,8 @@
-#elif defined(SDL_VIDEO_DRIVER_ANDROID) || defined(SDL_VIDEO_DRIVER_VIVANTE)
+#elif defined(SDL_VIDEO_DRIVER_ANDROID) || defined(SDL_VIDEO_DRIVER_VIVANTE) || defined(SDL_VIDEO_DRIVER_MALI)


diff --git a/src/video/SDL_sysvideo.h b/src/video/SDL_sysvideo.h
index 344efec48..000000000 100644
--- a/src/video/SDL_sysvideo.h
+++ b/src/video/SDL_sysvideo.h
@@ -478,6 +478,7 @@
 extern VideoBootStrap VIVANTE_bootstrap;
+extern VideoBootStrap MALI_bootstrap;


diff --git a/src/video/SDL_video.c b/src/video/SDL_video.c
index 04a1fdf65..000000000 100644
--- a/src/video/SDL_video.c
+++ b/src/video/SDL_video.c
@@ -78,6 +78,9 @@ static VideoBootStrap *bootstrap[] = {
 #ifdef SDL_VIDEO_DRIVER_VIVANTE
     &VIVANTE_bootstrap,
 #endif
+#ifdef SDL_VIDEO_DRIVER_MALI
+    &MALI_bootstrap,
+#endif


diff --git a/src/video/mali-fbdev/SDL_maliopengles.c b/src/video/mali-fbdev/SDL_maliopengles.c
new file mode 100644
--- /dev/null
+++ b/src/video/mali-fbdev/SDL_maliopengles.c
@@
+#include "../../SDL_internal.h"
+
+#if SDL_VIDEO_DRIVER_MALI && SDL_VIDEO_OPENGL_EGL
+
+#include "SDL_maliopengles.h"
+#include "SDL_malivideo.h"
+
+int MALI_GLES_LoadLibrary(_THIS, const char *path)
+{
+    return SDL_EGL_LoadLibrary(_this, path, EGL_DEFAULT_DISPLAY, 0);
+}
+
+SDL_EGL_CreateContext_impl(MALI)
+SDL_EGL_SwapWindow_impl(MALI)
+SDL_EGL_MakeCurrent_impl(MALI)
+
+#endif


diff --git a/src/video/mali-fbdev/SDL_maliopengles.h b/src/video/mali-fbdev/SDL_maliopengles.h
new file mode 100644
--- /dev/null
+++ b/src/video/mali-fbdev/SDL_maliopengles.h
@@
+#include "../../SDL_internal.h"
+
+#ifndef _SDL_maliopengles_h
+#define _SDL_maliopengles_h
+
+#if defined(SDL_VIDEO_DRIVER_MALI) && defined(SDL_VIDEO_OPENGL_EGL)
+
+#include "../SDL_sysvideo.h"
+#include "../SDL_egl_c.h"
+
+#define MALI_GLES_GetAttribute SDL_EGL_GetAttribute
+#define MALI_GLES_GetProcAddress SDL_EGL_GetProcAddress
+#define MALI_GLES_UnloadLibrary SDL_EGL_UnloadLibrary
+#define MALI_GLES_SetSwapInterval SDL_EGL_SetSwapInterval
+#define MALI_GLES_GetSwapInterval SDL_EGL_GetSwapInterval
+#define MALI_GLES_DeleteContext SDL_EGL_DeleteContext
+
+extern int MALI_GLES_LoadLibrary(_THIS, const char *path);
+extern SDL_GLContext MALI_GLES_CreateContext(_THIS, SDL_Window * window);
+extern int MALI_GLES_SwapWindow(_THIS, SDL_Window * window);
+extern int MALI_GLES_MakeCurrent(_THIS, SDL_Window * window, SDL_GLContext context);
+
+#endif
+#endif


diff --git a/src/video/mali-fbdev/SDL_malivideo.c b/src/video/mali-fbdev/SDL_malivideo.c
new file mode 100644
--- /dev/null
+++ b/src/video/mali-fbdev/SDL_malivideo.c
@@
+#include "../../SDL_internal.h"
+
+#if SDL_VIDEO_DRIVER_MALI
+
+#include "../SDL_sysvideo.h"
+#include "SDL_malivideo.h"
+#include "SDL_maliopengles.h"
+
+static void MALI_Destroy(SDL_VideoDevice *device)
+{
+    SDL_free(device);
+}
+
+static SDL_VideoDevice *MALI_Create()
+{
+    SDL_VideoDevice *device = SDL_calloc(1, sizeof(SDL_VideoDevice));
+
+    device->VideoInit = MALI_VideoInit;
+    device->VideoQuit = MALI_VideoQuit;
+    device->CreateWindow = MALI_CreateWindow;
+
+    device->GL_LoadLibrary = MALI_GLES_LoadLibrary;
+    device->GL_CreateContext = MALI_GLES_CreateContext;
+    device->GL_SwapWindow = MALI_GLES_SwapWindow;
+
+    device->free = MALI_Destroy;
+
+    return device;
+}
+
+VideoBootStrap MALI_bootstrap = {
+    "mali",
+    "Mali FBDEV",
+    MALI_Create,
+    NULL
+};
+
+#endif
